plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p5edges/orthogonal/OrthogonalRoutingGenerator.java

```diff
@@ -56,18 +59,47 @@
 
     /** differences below this tolerance value are treated as zero. */
     public static final double TOLERANCE = 1e-3;
+    
+    /** a special return value used by the conflict counting method. */
+    private static final int CRITICAL_CONFLICTS_DETECTED = -1;
 
     /** factor for edge spacing used to determine the {@link #conflictThreshold}. */
     private static final double CONFLICT_THRESHOLD_FACTOR = 0.5;
-    /** weight penalty for conflicts of horizontal line segments. */
-    private static final int CONFLICT_PENALTY = 16;
+    /** factor to compute {@link #criticalConflictThreshold}. */
+    private static final double CRITICAL_CONFLICT_THRESHOLD_FACTOR = 0.2;
```

Comment:
Any reasoning behind the factor? Or an expert guess ;)?

Reply:
The latter. This was determined experimentally during a bachelor thesis. I'll add a comment to that effect.